src/main.js

```diff
@@ -30,6 +30,11 @@ let mainWindow;
 let processIds = [];
 
 function createWindow() {
+  // Verify on opening if guppy is in the Applications Folder
+  if (!app.isInApplicationsFolder()) {
+    showMoveToApplicationsFolderDialog();
+  }
```

Comment:
There are a few conditions we shouldn't prompt on:

- If the app is running in development
- If the platform isn't MacOS
- The function exists (I believe that it always exists on MacOS, and never on any other platforms, but I could be wrong)
- If the user has previously ignored the option

The last one on that list is a bit tricky; we'd have to use electron-store, and I'd want to make sure it doesn't affect the other stuff we're doing with electron-store. So let's defer that one for now.

For the other ones, you should be able to do:

```js
const hasApplicationsFolder = process.platform === 'darwin' && typeof app.isInApplicationsFolder === "function";
const isProduction = process.env.NODE_ENV === 'production';

if (
  hasApplicationsFolder &&
  isProduction &&
  !app.isInApplicationsFolder()
) {
  ...
}
```

Reply:
Actually, I think these checks should move into the function as well, and the function should be named a bit more broadly:

```js
function createWindow() {
  manageApplicationLocation();
}

const manageApplicationLocation = () => {
  const hasApplicationsFolder = 
    process.platform === 'darwin' && 
    typeof app.isInApplicationsFolder === "function";

  const isProduction = process.env.NODE_ENV === 'production';

  if (!hasApplicationsFolder || !isProduction || app.isInApplicationsFolder()) {
    return;
  }

  dialog.showMessageBox(...)
}
```

My rationale is that `createWindow` is a very general function, and it shouldn't have to concern itself with all these conditions. This way, all the logic for application-moving is in 1 place